pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -0,0 +1,207 @@
+      "-Drenderer=${renderer}"
+    ]
+    ++ lib.mapAttrsToList (name: package: "-fsys=${name} --search-prefix ${lib.getLib package}") {
+      inherit glslang;
+    };
+
+  zigCheckFlags = finalAttrs.zigBuildFlags;
+
+  # Unit tests currently fail inside the sandbox
+  doCheck = false;
+
+  /**
+    Ghostty really likes all of it's resources to be in the same directory, so link them back after we split them
+
+    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/os/resourcesdir.zig#L11-L52
+    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L745-L750
+    - https://github.com/ghostty-org/ghostty/blob/4b4d4062dfed7b37424c7210d1230242c709e990/src/termio/Exec.zig#L818-L834
+
+    terminfo and shell integration should also be installable on remote machines
+
+    ```nix
+    { pkgs, ... }: {
+      environment.systemPackages = [ pkgs.ghostty.terminfo ];
+
+      programs.bash = {
+        interactiveShellInit = ''
+          if [[ "$TERM" == "xterm-ghostty" ]]; then
+            builtin source ${pkgs.ghostty.shell_integration}/bash/ghostty.bash
+          fi
+        '';
+      };
+    }
+    ```
+  */
+  postFixup = ''
+    ln -s $man/share/man $out/share/man
+
+    moveToOutput share/terminfo $terminfo
+    ln -s $terminfo/share/terminfo $out/share/terminfo
+
+    mv $out/share/ghostty/shell-integration $shell_integration
+    ln -s $shell_integration $out/share/ghostty/shell-integration
+
+    mv $out/share/vim/vimfiles $vim
+    rmdir $out/share/vim
+    ln -s $vim $out/share/vim-plugins
+
+
+    remove-references-to -t ${finalAttrs.deps} $out/bin/ghostty
+  '';
+
+  nativeInstallCheckInputs = [
+    versionCheckHook
+  ];
+
+  doInstallCheck = true;
+
+  versionCheckProgramArg = [ "--version" ];
+
+  passthru = {
+    tests = lib.optionalAttrs stdenv.hostPlatform.isLinux {
+      inherit (nixosTests) allTerminfo;
+      nixos = nixosTests.terminal-emulators.ghostty;
+    };
+  };
+
+  meta = {
+    description = "Fast, native, feature-rich terminal emulator pushing modern features";
+    longDescription = ''
+      Ghostty is a terminal emulator that differentiates itself by being
+      fast, feature-rich, and native. While there are many excellent terminal
+      emulators available, they all force you to choose between speed,
+      features, or native UIs. Ghostty provides all three.
+    '';
+    homepage = "https://ghostty.org/";
+    downloadPage = "https://ghostty.org/download";
+
+    license = lib.licenses.mit;
+    mainProgram = "ghostty";
+    maintainers = with lib.maintainers; [
+      jcollie
+      pluiedev
+      getchoo
+    ];
+    outputsToInstall = [
+      "out"
+      "man"
+      "shell_integration"
+      "terminfo"
+    ];
+    platforms = lib.platforms.linux ++ lib.platforms.darwin;
+    # Issues finding the SDK in the sandbox
+    broken = stdenv.hostPlatform.isDarwin;
```

Comment:
For this couldn't the thing directly be patched until zig global is patched.

Reply:
> For this couldn't the thing directly be patched until zig global is patched.

There would most likely be other changes required; it'd be best left to another PR

> https://github.com/NixOS/nixpkgs/pull/369080 was just merged, let's include it in this PR?

I'm not sure honestly. It lacks any testing (which I at least considered a blocker for stable support with the Linux version), the expression itself has a glaring issue in using a script based wrapper for no reason (this can affect startup time), and the postFixup is also a bit verbose. I'm not really sure why it was merged within an hour after no reviews :/

Reply:
> #369080 was just merged, let's include it in this PR?

I feel like it also lacks testing, wrapps two nix files for darwin and linux, but also approvals/reviewers from maintainers. I feel like we might need to revert the pr for the following issues before it hits `nixos-unstable`.

Reply:
As one of the maintainers, I feel that #369080 was merged *way* too hastily and in complete disregard to normal levels of procedural scrutiny that is expected in Nixpkgs. 

It was merged with so much haste that, in fact, the nix-owners CI didn't even *notify* the three original maintainers (including yours truly) about the PR's existence, and there's been essentially no feedback or polish done as a result. As such, I agree with a revert, and we need to do it more thoroughly and methodically next time.

Reply:
Opened https://github.com/NixOS/nixpkgs/pull/369529 to fix some of the issues. Doesn't seem like we can add testing currently due to Ghostty having issues running headlessly on Darwin, but it should improve things in the meantime

Maybe we can bring it in with 1.0.1